### ISSUE TITLE
Building with latest version of msbuild breaks for at least release/2.1 and release/2.2 branches

### ISSUE THREAD

**Issue by weshaggard** [dotnet/corefx#31925]
After updating VS and transitively msbuild to 15.8+ (15.8.166+gd4e8d81a88 is the version I currently have) msbuild started caring version 1.2.3.0 of System.Collections.Immutable with them which causes some our genfacades custom task to start failing.

We currently depend on 1.2.1.0 of System.Collections.Immutable and this mismatched version causes the following error while trying to convert pdbs.

```
 197:2>D:\git\corefx3\Tools\partialfacades.task.targets(64,5): error : Method not found: 'System.Collections.Immutable.ImmutableArray`1<Byte> System.Reflection.Metadata.DebugMetadataHeader.get_Id()'. [D:\git\corefx3\src\System.Buffers\src\System.Buffers.csproj]
```

This doesn't seem to affect master as we have already moved to the higher version of the dependency and they align so we might need to the same for the release branches.

**Comment by weshaggard** [comment]
Updating to the higher versions worked fine for the latest version of VS/msbuild but broke the earlier versions. To handle I'm trying a different approach which is to simply exclude System.Collections.Immutable from our task output that will allow us to use whichever version ships with msbuild. I've tested this on 15.7 and 15.8 and it works. I don't have a great way to test this on other versions but I think those 2 should be good enough.

**Comment by weshaggard** [comment]
This should be fixed in both the release/2.1 and 2.2 branches now.
